### DISCUSSION TITLE
Configs/Launchers working group

### DISCUSSION THREAD

**Discussion by JeanChristopheMorinPerso** [AcademySoftwareFoundation/rez#1772] [Proposals]
# Summary

The rez TSC wants to create a working group to work on configuring rez resolves and rez launchers.

One thing we (the maintainers and previous maintainers) have put a lot of efforts on making sure that rez is not used as a configuration system. We don't want studios to configure their shows using rez nor do we want them to use rez to manage their solves (say `rez env myshow`).

Rez is a package manager and doesn't really have a concept of managing environments. It has a concept of [suites](https://rez.readthedocs.io/en/stable/suites.html) but it is limitted and they are hard to manage and scale.

But we would still like to tackle issues related to configuration and launching softwares/environments. Though we think this should happen outside rez.

Here is some stuff that the working group could work on:
* Design and implement a pluggable layered configuration system to manage which packages should be used on a show/shot/asset, etc.
* Design and implement a generic and pluggable software launcher (CLI or GUI or both).
* Create guidelines, documentations and best practices on how to manage rez environments.
* Integration guides.
* Research and suggest possible workflows and existing tools that could be re-used or used as a base.

These are just examples. The scope will need to be defined, including the expected outcome.

# Why?

So why a working group? After talking with other ASWF projects, I think a working group would be a good fit for the purpose. The process is described in https://tac.aswf.io/process/tsc_faq.html#q-could-a-project-start-working-groups-or-sub-committees-within-their-project. That document states four questions that we should ask ourselves before starting a new working group:

> Is this initiative big enough, with a long enough timeline (6+ months usually, or ongoing is also ok)

I think the effort will span more than 6 months.

> Will this initiative require a level of collaboration and discussion that is not possible in the general TSC meeting?

Definitely. It will include a lot of discussions on pipeline management and designs and tooling and will require collaboration from a lot of people. The TSC members don't have the bandwidth to manage all this.

> Should this initiative involve folks not on the TSC?

Yes. As said above, the TSC members don't have the bandwidth to handle the project as a whole and we will need help from non TSC members to lead this.

> Is this a high priority for the project?

Yes. All studios (even studios not using rez) face the same problems and it's a big barrier of entry for rez users.

So with this said, we are looking for help! We are looking for people that are interested in:
* Lead the project(s)
* Project management
* Participate in any preparation work
    * Research on prior art
    * Documenting what studios do today
    * Document requirements
    * Etc
* Participate in designing solutions
* Implement solutions that the group based on designs made by the group
* Test
* Integrate into studios
* Give feedback

Note that we are looking for people that will commit time and energy on this effort.

# Immediate tasks

This is a list of things that have to be done before any working can start.

1. Identify interested parties (individuals and studios).
2. Create working group charter (which includes defining the scope).

Once a chartr is in place, we could start to look at requirements, do research, etc.

# Interested?

Are you interested and ready to commit time into this effort? if so, please reach out to us by either commenting in this discussion or through Slack. If you already know what kind of help you can provide, please let us know.

We are also looking for feedback and ideas.

Also, please spread the word! Talk to your friends and colleagues and the tech leaders in your studio. The more voices and opinions we have the better. We need people with all type of experiences and levels.

Thanks, the rez TSC.

**Comment by MrLixm** [comment]
Thanks for writing this down. Looking forward to contributing as an individual.
